spec/features/editing_topics/show_topics_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+RSpec.feature "Show the topics for a document" do
+  scenario do
+    given_there_is_a_document
+    when_the_document_has_no_topics
+    and_i_visit_the_document_page
+    then_i_see_there_are_no_topics
+
+    when_the_document_has_a_topic
+    and_i_visit_the_document_page
+    then_i_see_the_topic_breadcrumb
+  end
+
+  def given_there_is_a_document
+    @document = create :document
+  end
+
+  def when_the_document_has_no_topics
+    publishing_api_has_links(
+      "content_id" => @document.content_id,
+      "links" => {},
+    )
+  end
+
+  def and_i_visit_the_document_page
+    visit document_path(Document.last)
+  end
+
+  def then_i_see_there_are_no_topics
+    expect(page).to have_content(I18n.t("documents.show.topics.no_topics"))
+  end
+
+  def when_the_document_has_a_topic
+    publishing_api_has_links(
+      "content_id" => @document.content_id,
+      "links" => {
+        "taxons" => %w(level_three_topic),
+      },
+    )
+
+    # GOV.UK homepage
+    publishing_api_has_expanded_links(
+      "content_id" => "f3bbdec2-0e62-4520-a7fd-6ffd5d36e03a",
+      "expanded_links" => {
+        "level_one_taxons" => [
+          {
+            "title" => "Level One Topic",
+            "content_id" => "level_one_topic",
+          },
+        ],
+      },
+    )
+
+    publishing_api_has_expanded_links(
+      "content_id" => "level_one_topic",
+      "expanded_links" => {
+        "child_taxons" => [
+          {
+            "content_id" => "level_two_topic",
+            "title" => "Level Two Topic",
+            "links" => {
+              "child_taxons" => [
+                {
+                  "content_id" => "level_three_topic",
+                  "title" => "Level Three Topic",
+                  "links" => {},
+                },
+              ],
+            },
+          },
+        ],
+      },
+    )
```

Comment:
Is this the dreaded second instance of this or is this the same test?

Reply:
This is the dreaded second instance ;-). I think it will also be the last, though, and I'd like to avoid refactoring it without more evidence - the two specs are very isolated, after all.

Reply:
Ok - I'll trust you to do the responsible thing on the third time 😉

Reply:
Thanks. I suppose it may not be me doing it, but I expect one or other of us'll be reviewing it!